Gordon360/Services/ProfileService.cs

```diff
@@ -282,6 +283,143 @@ public async Task UpdateProfileLinkAsync(string username, string type, CUSTOM_PR
+        /// <param name="currentUserType">personnel type of the logged-in user</param>
+        /// <param name="stu">original profile of the student being searched</param>
+        /// <returns>public profile of the student based on individual privacy settings</returns>
+        public PublicStudentProfileViewModel ToPublicStudentProfileViewModel
+            (string username, string currentUserType, StudentProfileViewModel stu)
+        {
+            PublicStudentProfileViewModel publicStu = (PublicStudentProfileViewModel)stu;
+            var account = _accountService.GetAccountByUsername(username);
+
+            // select all privacy settings
+            var privacy = _context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID);
+            // get type of viewmodel for reflection property setting
+            Type s_vm = new PublicStudentProfileViewModel().GetType();
+
+            foreach (UserPrivacy_Fields field in _context.UserPrivacy_Fields)
+            {
+                var row = privacy.FirstOrDefault(p => p.Field == field.Field);
+                // HomePhone and SpouseName fields are not for students
+                if (field.Field == "HomePhone" || field.Field == "SpouseName") { }
+                else
+                {
+                    if (row is UserPrivacy_Settings instance)
+                    {
+                        if (instance.Visibility == "Private" || (instance.Visibility == "FacStaff" && currentUserType != "fac"))
+                        {
+                            s_vm.GetProperty(field.Field).SetValue(publicStu, "Private as requested.");
+                        }
+                    }
+                    else
+                    {
+                        s_vm.GetProperty(field.Field).SetValue(publicStu, "Private as requested.");
+                    }
+                }
+            }
+
+            return publicStu;
+        }
+
+        /// <summary>
+        /// get privacy setting for particular user
+        /// </summary>
+        /// <param name="username">AD username</param>
+        /// <returns></returns>
+        public IEnumerable<UserPrivacyViewModel> GetPrivacySettingAsync(string username)
+        {
+            var account = _accountService.GetAccountByUsername(username);
+
+            // select all privacy settings
+            var privacy = _context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID);
+
+            if (privacy == null)
+            {
+                return null;
+            }
+
+            List<UserPrivacyViewModel> resultList = new();
+
+            foreach (UserPrivacy_Fields field in _context.UserPrivacy_Fields)
+            {
+                var row = privacy.FirstOrDefault(p => p.Field == field.Field);
+                if (row is UserPrivacy_Settings instance)
+                {
+                    resultList.Add(new UserPrivacyViewModel(field.Field, instance.Visibility));
```

Comment:
```suggestion
                    resultList.Add(new UserPrivacyViewModel(field.Field, instance.Visibility));
```
Look at some of the other viewmodels that we have and look at how we implicitly cast. `Implicit Operators` are something you should look into.